app/models/book_csv_importer.rb

```diff
@@ -0,0 +1,62 @@
+class BookCsvImporter
+  def initialize(csv_file_content)
+    @csv_file_content = csv_file_content
+  end
+
+  def import!
+    csv_table = CSV.parse(@csv_file_content, headers: true)
+
+    if csv_table.empty?
+      raise "The file is empty"
+    end
+
+    if csv_table.headers.join(",") != "Name,ISBN,Authors,Genres,Featured,Published At,"
```

Comment:
The question this boolean answers is

> Are the headers in the CSV the same and in the same order as this string?

but this has a downside in that it is too  restrictive. In reality, we don't care what orders the columns are in.

The question we want to answer is

> Does the CSV contain all the headers we require?

`csv_table.headers` is an array of strings, where each string is the name of a column in the CSV.

Can you think of a away to compare that array to an object, that will allow us to make an order-independent comparison?

As a hint: It's almost always only going to be possible to compare two things of the same class (EG: Array and Array, Hash and Hash)

The methods you need can be found here: https://ruby-doc.org/core-3.0.0/Array.html or potentially here: https://ruby-doc.org/stdlib-2.7.1/libdoc/set/rdoc/Set.html

This is probably a Google-able answer, too, which is how I would have worked out how to solve this many moons ago